### PR TITLE
Chart tooltips

### PR DESCRIPTION
**Pull request by AdrianLxM**
If you touch a value point in the main chart, it will show a small toast with the time and value.

### CODE REVIEW

File: app/src/main/java/com/eveningoutpost/dexdrip/Home.java [hunk 1]
```diff
@@ -210,6 +210,7 @@ public void setColorFilter(ColorFilter cf) {}
         previewChart.setZoomType(ZoomType.HORIZONTAL);
 
         chart.setLineChartData(bgGraphBuilder.lineData());
+        chart.setOnValueTouchListener(bgGraphBuilder.getOnValueSelectTooltipListener(this));
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Home.java]: Passing around activities is a good way to leak large amounts of RAM.  I would feel better about it if it instantiated a new anonymous listener and passed that instead.

File: app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+import android.app.Activity;
 import android.content.Context;
 import android.content.SharedPreferences;
 import android.content.res.Configuration;
 import android.graphics.Color;
 import android.preference.PreferenceManager;
 import android.text.format.DateFormat;
+import android.widget.Toast;
 
+import com.eveningoutpost.dexdrip.Home;
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java]: This import is unused.

File: app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java [hunk 3]
```diff
@@ -408,4 +413,36 @@ public String unit() {
         }
 
     }
+
+    public OnValueSelectTooltipListener getOnValueSelectTooltipListener(Activity activity){
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java]: After a little more thought about this, can you do this instead?  I don't think a new Application context is necessary for a toast.

```
    public OnValueSelectTooltipListener getOnValueSelectTooltipListener(){
        return new OnValueSelectTooltipListener();
    }

    public class OnValueSelectTooltipListener implements LineChartOnValueSelectListener{

        private Toast tooltip;

        @Override
        public synchronized void onValueSelected(int i, int i1, PointValue pointValue) {
            final java.text.DateFormat timeFormat = DateFormat.getTimeFormat(context);
            //Won't give the exact time of the reading but the time on the grid: close enough.
            Long time = ((long)pointValue.getX())*FUZZER;
            if(tooltip!= null){
                tooltip.cancel();
            }
            tooltip = Toast.makeText(context, timeFormat.format(time)+ ": " + Math.round(pointValue.getY()*10)/ 10d , Toast.LENGTH_LONG);
            tooltip.show();
        }

        @Override
        public void onValueDeselected() {
            // do nothing
        }
    }
```

**Comment by AdrianLxM** [comment]
I changed it how you proposed it @mgranberry.
I don't think it would affect the RAM at all (maybe GC load minimal) as we already hold a reference to said activity. But simpler is better - no need for redundant references.

**Comment by mgranberry** [comment]
I agree.  It would only be 8 bytes.  But code has a way of getting copied
in this repo and it's best to keep it clean wherever possible.

**Pull request merged**